Individual Project Planning Stage

This is Simran Heir's notebook for the individual planning stage.
I have choosen Broad Question 1: What player characteristics and behaviours are most predictive of subscribing to a game-related newsletter, and how do these features differ between various player types? Can played_hours and experience predict subscription to a game related newsletter in a dataset?

I am going to load tidyverse into the R notebook.

In [1]:
library(tidyverse)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


loading player data

In [2]:
url <- "https://raw.githubusercontent.com/SimranHeir/DSCI-100-Project-Planning-Stage-Individual-Simran-Heir/refs/heads/main/players.csv"
player_data <- read_csv (url)


Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


In [3]:
head (player_data, 6)

experience subscribe
1 Pro         TRUE    
2 Veteran     TRUE    
3 Veteran    FALSE    
4 Amateur     TRUE    
5 Regular     TRUE    
6 Amateur     TRUE    
  hashedEmail                                                      played_hours
1 f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d 30.3        
2 f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9  3.8        
3 b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28  0.0        
4 23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5  0.7        
5 7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e  0.1        
6 f58aad5996a435f16b0284a3b267f973f9af99e7a89bee0430055a44fa92f977  0.0        
  name      gender Age
1 Morgan    Male    9 
2 Christian Male   17 
3 Blake     Male   17 
4 Flora     Female 21 
5 Kylie     Male   21 
6 Adrian    Female 17

How many observations are there for player data?

In [4]:
nrow (player_data)

[1] 196

Player summary statistics

In [5]:
summary_player <- summary (player_data)
summary_player

  experience        subscribe       hashedEmail         played_hours    
 Length:196         Mode :logical   Length:196         Min.   :  0.000  
 Class :character   FALSE:52        Class :character   1st Qu.:  0.000  
 Mode  :character   TRUE :144       Mode  :character   Median :  0.100  
                                                       Mean   :  5.846  
                                                       3rd Qu.:  0.600  
                                                       Max.   :223.100  
                                                                        
     name              gender               Age       
 Length:196         Length:196         Min.   : 9.00  
 Class :character   Class :character   1st Qu.:17.00  
 Mode  :character   Mode  :character   Median :19.00  
                                       Mean   :21.14  
                                       3rd Qu.:22.75  
                                       Max.   :58.00  
                               

The data is tidy because it is organized and meets the requirments for being tidy

loading session data

In [6]:
url <- "https://raw.githubusercontent.com/SimranHeir/DSCI-100-Project-Planning-Stage-Individual-Simran-Heir/refs/heads/main/sessions.csv"
sessiondata <- read_csv (url)


Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


In [7]:
head (sessiondata, 6)

hashedEmail                                                     
1 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
2 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
3 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
4 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
5 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
6 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
  start_time       end_time         original_start_time original_end_time
1 30/06/2024 18:12 30/06/2024 18:24 1.71977e+12         1.71977e+12      
2 17/06/2024 23:33 17/06/2024 23:46 1.71867e+12         1.71867e+12      
3 25/07/2024 17:34 25/07/2024 17:57 1.72193e+12         1.72193e+12      
4 25/07/2024 03:22 25/07/2024 03:58 1.72188e+12         1.72188e+12      
5 25/05/2024 16:01 25/05/2024 16:12 1.71665e+12         1.71665e+12      
6 23/06/2024 15:08 23/06/2024 17:10 1.71916e+12         1.71916e+12

How many observations are there for session data?

In [8]:
nrow(sessiondata)

[1] 1535

In [9]:
summary_session <- summary (sessiondata)
summary_session

 hashedEmail         start_time          end_time         original_start_time
 Length:1535        Length:1535        Length:1535        Min.   :1.712e+12  
 Class :character   Class :character   Class :character   1st Qu.:1.716e+12  
 Mode  :character   Mode  :character   Mode  :character   Median :1.719e+12  
                                                          Mean   :1.719e+12  
                                                          3rd Qu.:1.722e+12  
                                                          Max.   :1.727e+12  
                                                                             
 original_end_time  
 Min.   :1.712e+12  
 1st Qu.:1.716e+12  
 Median :1.719e+12  
 Mean   :1.719e+12  
 3rd Qu.:1.722e+12  
 Max.   :1.727e+12  
 NA's   :2          

The data is not tidy and needs to be reorganized.

Activity #1: Explore Datasets - Preliminary
what is the dataset about?
how many variables are there?
how many observations are there?
write a short description

The data is about information about players that play video games, the duration of playing video games, the player email, and whether they would subscribe to a game related newsletter.
Player data:
The data is about information about the players, the time played, the experience, and whether they have subscribed.
There are 7 variables: experience, subscribe, hashEmail, player_hours, name, gender, and age.
There are 196 observations for this dataset.

Session data:
The data is about the session. It includes the email, date and time played.
There are 5 variables: hashEmail, start_time, end_time, original_start_time, and original_end_time.
There are 1535 observations for this dataset.

Reorganizing data

In [10]:
organizing_start_data <- sessiondata |>
    separate (start_time, into = c("start_date", "start_time"), sep = " ")

organizing_end_data <- organizing_start_data |>
    separate (end_time, into = c("end_date", "end_time"), sep = " ")

Reorganizing original time column

Reorganize

In [11]:
original_time_hours <- organizing_end_data |>
    mutate (original_start_time_hours = original_start_time/3600000, original_end_time_hours = original_end_time/3600000)

What to change the start time and end time to just average time, so that I can combine the data sets.

In [12]:
session_data <- original_time_hours |>
    mutate (played_hours = (sqrt ((original_end_time-original_start_time)^2)))

In [13]:
head (session_data, 6)

hashedEmail                                                      start_date
1 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf 30/06/2024
2 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686 17/06/2024
3 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc 25/07/2024
4 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf 25/07/2024
5 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686 25/05/2024
6 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf 23/06/2024
  start_time end_date   end_time original_start_time original_end_time
1 18:12      30/06/2024 18:24    1.71977e+12         1.71977e+12      
2 23:33      17/06/2024 23:46    1.71867e+12         1.71867e+12      
3 17:34      25/07/2024 17:57    1.72193e+12         1.72193e+12      
4 03:22      25/07/2024 03:58    1.72188e+12         1.72188e+12      
5 16:01      25/05/2024 16:12    1.71665e+12         1.71665e+12      
6 15:08      23/06/2024 17:10    1.71916e+12         1.71916e+12      
  original_start_time_hours original_end_time_hours played_hours
1 477713.9                  477713.9                0           
2 477408.3                  477408.3                0           
3 478313.9                  478313.9                0           
4 478300.0                  478300.0                0           
5 476847.2                  476847.2                0           
6 477544.4                  477544.4                0

Combining the two data sets (player data + session data)

using semi_join to combine the player data and session data

In [14]:
joined_data <- semi_join (player_data, session_data)
joined_data

Joining with `by = join_by(hashedEmail, played_hours)`


experience subscribe
1  Veteran    FALSE    
2  Amateur    FALSE    
3  Regular     TRUE    
4  Amateur     TRUE    
5  Amateur     TRUE    
6  Amateur    FALSE    
7  Beginner   FALSE    
8  Beginner    TRUE    
9  Beginner    TRUE    
10 Regular    FALSE    
11 Amateur     TRUE    
12 Amateur     TRUE    
13 Regular    FALSE    
14 Amateur    FALSE    
   hashedEmail                                                     
1  b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28
2  1d2371d8a35c8831034b25bda8764539ab7db0f63938696917c447128a2540dd
3  e21a324ccf5c873bafe82e47d5137b36aa312ee4803eeb5cbc6cf945f9bb3f38
4  dbb20960cd4db4900dde7832e349dae46f7688583c8138b34785d9c7e509b5b9
5  25879aecc205544bc6505f9faf768356e0a3b7126057308ae3235f8a514e9a62
6  2917e2dacafed92d017b9f356ce90a442bea6e4ad59dbbb484a920398acb8974
7  722ab6d91280c439f2780767a1250889e93f75fc0ba0bc48175da022ab3e9ea9
8  ed3e6d043dee74fad932be44d0600f4502055198c511927b0c906f50b57d9582
9  7b5c323f9a9e21b6bcc2a8c431e96f9f2155c7b4318a68d3e7d9edd2018095a9
10 a2a0612e9a7da558cbac2ee3c816740324505a69a6e042c94aa1b67731f9ff01
11 2cfed571797b66cc810c32562fc5b0f70b5bec0f5250799245527135bbf9662e
12 bda1905e54b6c745bcced9d59ce655a5bfd03c35cc6abdd8d1610e5c416f79d1
13 7b17513f11968feaf6b02977885c436dc2a4549489ca25ef03022f9fbae19d81
14 d572f391d452b76ea2d7e5e53a3d38bfd7499c7399db299bd4fedb06a46ad5bb
   played_hours name      gender            Age
1  0            Blake     Male              17 
2  0            Emerson   Male              21 
3  0            Faye      Male              17 
4  0            Jia       Female            26 
5  0            Eli       Female            17 
6  0            Haider    Two-Spirited      26 
7  0            Leyla     Female            17 
8  0            Hamish    Male              17 
9  0            Xenos     Two-Spirited      17 
10 0            Cornelius Male              18 
11 0            Milo      Male              16 
12 0            Vaughn    Male              17 
13 0            Balthazar Male              50 
14 0            Dylan     Prefer not to say 57

Activity #2: Explore Dataset Part 2 - Outcome Variable
Identify the main outcome/catergory/label variable?
How many values/groups are in this variable?
How many observations are there in each group?


Activity #3: Explore Datasets Part 3 - Visualisations
Make some visualisations of the outcome variable:
what does the distribution of the variable look like
what relationship does it have with some of the other variables?